ksql-engine/src/test/java/io/confluent/ksql/EndToEndEngineTest.java

```diff
@@ -0,0 +1,554 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.confluent.ksql;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.serializers.KafkaAvroDeserializer;
+import io.confluent.kafka.serializers.KafkaAvroSerializer;
+import io.confluent.ksql.function.InternalFunctionRegistry;
+import io.confluent.ksql.serde.DataSource;
+import io.confluent.ksql.util.KsqlConstants;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.TopologyTestDriver;
+import org.apache.kafka.streams.kstream.TimeWindowedDeserializer;
+import org.apache.kafka.streams.kstream.Windowed;
+import org.apache.kafka.streams.kstream.internals.TimeWindow;
+import org.apache.kafka.streams.test.ConsumerRecordFactory;
+import org.apache.kafka.streams.test.OutputVerifier;
+import org.apache.kafka.test.TestUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import io.confluent.kafka.schemaregistry.client.MockSchemaRegistryClient;
+import io.confluent.ksql.metastore.MetaStore;
+import io.confluent.ksql.metastore.MetaStoreImpl;
+import io.confluent.ksql.util.FakeKafkaTopicClient;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.QueryMetadata;
+
+@Ignore
+public class EndToEndEngineTest {
+
+  private final MetaStore metaStore = new MetaStoreImpl(new InternalFunctionRegistry());
+  private final Map<String, Object> config = new HashMap<String, Object>() {{
+    put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+    put("application.id", "KSQL-TEST");
+    put("commit.interval.ms", 0);
+    put("cache.max.bytes.buffering", 0);
+    put("auto.offset.reset", "earliest");
+    put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getPath());
+  }};
+  private final Properties streamsProperties = new Properties();
+  private SchemaRegistryClient schemaRegistryClient = new MockSchemaRegistryClient();
```

Comment:
can this be final?

Reply:
yeah there are a few other places I need to add final as well.